chemprop/data/utils.py

```diff
@@ -333,98 +373,166 @@ def get_data(path: str,
         gt_targets, lt_targets = None, None
 
     # Load data
-    with open(path) as f:
+    extension = os.path.splitext(path)[1]
+    if extension == '.csv':
+        f = open(path)
         reader = csv.DictReader(f)
         fieldnames = reader.fieldnames
-        if any([c not in fieldnames for c in smiles_columns]):
-            raise ValueError(f'Data file did not contain all provided smiles columns: {smiles_columns}. Data file field names are: {fieldnames}')
-        if any([c not in fieldnames for c in target_columns]):
-            raise ValueError(f'Data file did not contain all provided target columns: {target_columns}. Data file field names are: {fieldnames}')
+    elif extension in ['.pkl', '.pckl', '.pickle']:
+        reader = pd.read_pickle(path)
+        fieldnames = reader.columns.tolist()
+        reader = reader.iterrows()
+
```

Comment:
After the `if csv` block and the `elif pkl` block, you should have a `else: raise NotImplementedError` block saying that the extension isn't a supported one.